wave_lang/kernel/wave/execution_engine/buffer_utils.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2025 The IREE Authors
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+
+#include <Python.h>
+#include <cstdint>
+
+/// StridedMemRefType is the descriptor structure used by MLIR for memrefs.
+/// This matches the ABI used by MLIR's memref lowering.
+template <typename T, int N> struct StridedMemRefType {
+  T *basePtr;         // Pointer to the allocated buffer
+  T *data;            // Aligned data pointer
+  int64_t offset;     // Offset in elements
+  int64_t sizes[N];   // Size of each dimension
+  int64_t strides[N]; // Stride of each dimension in elements
+};
+
+/// Rank-1 memref descriptor for memref<?xi8>
+using MemRef1Di8 = StridedMemRefType<uint8_t, 1>;
```

Comment:
I think I saw at least comments about rank-0 memrefs above, but can't find them quickly given the size of the PR. Please double-check that we consistently use either rank-0 or rank-1 memrefs.